dotnet/src/Microsoft.Agents.AI.DevUI/Conversations/Models/OpenAIContentConverter.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.Extensions.AI;
+
+namespace Microsoft.Agents.AI.Hosting.OpenAI.Conversations.Models;
+
+/// <summary>
+/// JSON converter for lists of AIContent that produces OpenAI-compatible format.
+/// </summary>
+[SuppressMessage("CodeQuality", "CA1812:Avoid uninstantiated internal classes", Justification = "Instantiated via reflection by JSON serialization.")]
+internal sealed class OpenAIContentListConverter : JsonConverter<IList<AIContent>>
+{
+    private static readonly OpenAIContentConverter s_itemConverter = new();
+
+    public override IList<AIContent>? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+    {
+        if (reader.TokenType != JsonTokenType.StartArray)
+        {
+            throw new JsonException($"Expected StartArray token, got {reader.TokenType}");
+        }
+
+        var list = new List<AIContent>();
+        while (reader.Read())
+        {
+            if (reader.TokenType == JsonTokenType.EndArray)
+            {
+                return list;
+            }
+
+            var item = s_itemConverter.Read(ref reader, typeof(AIContent), options);
+            if (item is not null)
+            {
+                list.Add(item);
+            }
+        }
+
+        throw new JsonException("Unexpected end of JSON");
+    }
+
+    public override void Write(Utf8JsonWriter writer, IList<AIContent> value, JsonSerializerOptions options)
+    {
+        writer.WriteStartArray();
+        foreach (var item in value)
+        {
+            s_itemConverter.Write(writer, item, options);
+        }
+        writer.WriteEndArray();
+    }
+}
+
+/// <summary>
+/// JSON converter for AIContent that produces OpenAI-compatible format.
+/// Converts TextContent to {"type": "input_text", "text": "..."} format.
+/// </summary>
+#pragma warning disable CA1812 // Avoid uninstantiated internal classes - This is used via the JsonConverter attribute
```

Comment:
Seems like we need to update CA1812 to be aware of more ways types can be used than just instantiating them